ecopod/sendRawTransactionConditional.md

```diff
@@ -0,0 +1,53 @@
+# Purpose
+
+To fully unlock ERC-4337 on the OP-Stack.
+
+# Summary
+
+By providing an auxilliary transaction submission mechanism, [eth_sendRawTransactionConditional](https://notes.ethereum.org/@yoav/SkaX2lS9j), we can enable Bundlers to submit Entrypoint transactions with stronger guarantees, avoiding costly inadvertent reverts.
+
+We can implement this endpoint in op-geth with an additional layer of external validation to ensure this endpoint is safe from DoS attacks, does not supersede `eth_sendRawTransaction`, and can be deprecated if native account abstraction efforts in the future absolves the need for this endpoint.
+
+**Note:** As a result of [EIP-3074](https://eips.ethereum.org/EIPS/eip-3074), the interop project is also exploring external validation as a mechanism to preventing DoS attack. A low-stakes scoped version of this for `eth_sendRawTransactionConditional` is a good opportunity for learning.
+
+
+# Problem Statement + Context
+
+Bundlers aggregate UserOps from an external mempool into a single transaction sent to the enshrined 4337 `Entrypoint` contract. It is the Bundler's responsibility to ensure every UserOp in the transaction will not revert when its validation step is executed. In between tx submission and inclusion, there is a chance network state changes causes the transaction to revert, even after local bundler validation.
+
+With private Bundler mempools, the likelihood of these reverts are small. However shared 4337 mempools are coming into [production](https://medium.com/etherspot/decentralized-future-erc-4337-shared-mempool-launches-on-ethereum-b6c860072f41), launched on Ethereum and some L2 testnets, decentralizing 4337 infrastructure. This is possible with no change on Ethereum through special block builders like Flashbots which support atomic transaction inclusion. An endpoint like `eth_sendRawTransactionConditional` is required to launch a shared mempools on L2 as there's only a single block builder, the sequencer, and the likelihood of reverts with a shared mempool would be too high of a cost for bundlers to operate.
+
+
+# Alternatives Considered
+
+## Continue with private 4337 mempools
+
+4337 account abstraction is live on optimism. Dapps likely utilize the bundler endpoints that are come with a vendor-specific SDK -- Alchemy, Pimilico, Thirdweb, etc. As 4337 infrastructure becomes more permissionless, we will likely later have to play catch-up to ensure the OP-Stack remains compatible while other L2 offerings have already moved towards supporting `eth_sendRawTransactionConditional`.
+
+## Verticalize the OP-Stack
+
```

Comment:
Related to this paragraph, there is also a solution where the sequencer runs its own 4337 mempool software that somebody else maintains, ie not make it part of the op stack itself but instead part of our infrastructure. Are there any benefits to this sort of architecture?

Reply:
So this is included as apart of the proposed solution. By having `eth_sendRawTransactionConditional` be authenticated, the sequencer can choose to only allow their own bundler in the allowlist

> A chain operator can still choose to verticalize in proposed solution solution by allowlisting only its own bundler through `eth_sendRawTransactionConditional`, achieving the same outcome.